### ISSUE TITLE
`dvc diff`: unexpected output in Github Actions

### ISSUE THREAD

**Issue by yathomasi** [iterative/dvc#8875]
# Bug Report

The main issue is the `dvc diff` between the current branch and base branch in the PR on GHA is not generating the expected results.
The same command `dvc diff origin/test-dvc-changes-report-no-deploy origin/add-dvc-changes-report` works fine locally but not in GHA. I did try `dvc pull --all-branches` on GHA before `diff` but still the same unexpected result.

### Reproduce

Related PR with script: https://github.com/iterative/iterative.ai/pull/759
Test PR: https://github.com/iterative/iterative.ai/pull/760

### Expected

![image](https://user-images.githubusercontent.com/20840228/214481154-3b3d3a68-ee51-4d9e-8c2b-55399bcc27a8.png)

**Comment by skshetry** [comment]
Duplicate of https://github.com/iterative/dvc/issues/7661 and https://github.com/iterative/dvc-data/issues/83?

**Comment by pmrowla** [comment]
Im not sure this is the same, since using `dvc pull -a` before doing the diff should have fetched the required objects into cache. The issue here is that using `dvc pull -a` before the diff gives the expected result when testing locally, but not in GHA CI runs.

**Comment by pmrowla** [comment]
@yathomasi it looks like this is fixed by @dberenbaum's suggestion? (issue is that `origin/...` refs aren't automatically tracked as local branches in GHA and will be excluded in `dvc pull -a`)

**Comment by yathomasi** [comment]
Yes, it looks to be fixed by @dberenbaum suggestion. ([output looks as expected here](https://github.com/iterative/iterative.ai/pull/760#issuecomment-1401633911) )

> issue is that `origin/...` refs aren't automatically tracked as local branches in GHA and will be excluded in `dvc pull -a`

Yes, this seems to be the case. So, do we have better way for remote refs? 
And, I guess we at least want to document it somewhere maybe on troubleshooting

**Comment by pmrowla** [comment]
Closing this as expected behavior (see discussion in https://github.com/iterative/iterative.ai/pull/759#issuecomment-1408133016). If there are no local branches `--all-branches` doesn't do anything, if you are dealing with purely remote branches in CI (or in a fresh local clone) you need to `git checkout` the `origin/...` remote ref names you want first so that git actually creates local branches.
